matplotlib2tikz/legend.py

```diff
@@ -106,44 +107,74 @@ def draw_legend(data, obj):
 
     if loc == 1:
         # upper right
-        position = None
+        if obj._bbox_to_anchor:
```

Comment:
Instead of 10 `if`-`else` clauses, one big `if`-`else` would probably look cleaner. That removes the risk of accidentally changing the clause for just one `loc`, too.